### PR TITLE
Missing prop bug fix

### PR DESCRIPTION
**Pull request by AnyaVeselova**
# Description

**Relates #[[#366](https://github.com/enBloc-org/kindly/issues/366)] **  

### Files changed

 - AddNewItemForm.tsx (deletes onSubmit; refactors the form using react-hook-form library; extracts inputs into separate components; handles error with errors object form formState) and a separate custom component (ErrrMessage.tsx)

### CODE REVIEW

**Comment by AnyaVeselova** [comment]
Those errors come from UploadImageInput.tsx. 
Screenshot from 2024-10-30 19-37-33
Screenshot from 2024-10-30 19-38-10
Screenshot from 2024-10-30 19-38-00

**Comment by AnyaVeselova** [comment]
Update: creating images bucket in storage in db solved the issue with images upload

**Comment by AnyaVeselova** [comment]
tests fail despite me being able to upload a new item and delete it

**Comment by camelPhonso** [comment]
> tests fail despite me being able to upload a new item and delete it

Hi Anna, I can see you're refactored the code to break out some elements into new components. It's quite possible that the tests cannot find the correct elements now and create a false fail. Please have a look through the logs and let's see if there's a refactor that needs to happen there

File: components/form/UploadImageInput.tsx [hunk 1]
```diff
@@ -10,117 +10,90 @@ type UploadImageProps = {
   setError?: (error: string) => void;
   isRequired?: boolean;
   imageType: 'item' | 'profile';
+  name: string;
 };
 
 const CDN = `${process.env.NEXT_PUBLIC_SUPABASE_URL}/storage/v1/object/public/images/`;
```
Comment (nichgalzin) [inline components/form/UploadImageInput.tsx:17]: The scope of this refactor is really outside the scope of this issue and removes some functionality that we don't want to lose, like removing a user profile picture they already have one. This component is already scheduled for a refactor, see issue #334. Please revert the changes on this file.

File: components/hooks/useMediaQuery.ts [hunk 2]
```diff
@@ -4,26 +4,20 @@ const useMediaQuery = (width: number) => {
   const [targetReached, setTargetReached] = useState(false);
 
   const updateTarget = useCallback((e: MediaQueryListEvent) => {
```
Comment (nichgalzin) [inline components/hooks/useMediaQuery.ts:6]: This is all fine and it does all make the code tidier. But please to keep PR's focused and concise to the issue. If you see things you want to refactor, please submit a issue request and we are happy to make that for you and you can pick it up.

File: components/form/SizeSelector.tsx [hunk 3]
```diff
@@ -1,13 +1,17 @@
 import React, { useEffect, useState } from 'react';
 import { UseFormRegister } from 'react-hook-form';
-import { PartialItem } from '@/types/supabaseTypes';
 
 interface SizeSelectorProps {
   category: string;
-  register: UseFormRegister<PartialItem>;
+  register: UseFormRegister<any>;
```
Comment (nichgalzin) [inline components/form/SizeSelector.tsx:6]: To keep or types strong we don't want to use <any> as type. So please either revert, if you're having trouble getting the types to work. Let us know

**Review by nichgalzin — CHANGES_REQUESTED** [review]
It seems like two of the tests related to items are failing, this may be because the tests are looking for elements or attributes that are looking for may have been moved in your refactors. 
    We really appreciate the enthusiasm you are being to the project, but as this project has grow we need it to move slowly and carefully so as not to break as many things as we fix. It's also good practice to keep changes small and focused so that they are easy to review and see there not on effect. 
    For this PR I've requested some changes and please see update the test to get them to run.

File: components/form/ErrorMessage.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  customMessage,
+}) => {
+  return (
+    <div className='error-message-container'>
+      <ErrorMessage
+        errors={errors}
+        name={name}
+        render={({ message }) => <p className='error-message'>{message}</p>}
+      />
+      {!errors[name] && customMessage && (
```
[the remark this answers is not in the document]
Comment (AnyaVeselova) [reply components/form/ErrorMessage.tsx:22]: Thank you so much for the valuable feedback @nichgalzin ! I realize I got a bit carried away with the changes. I’ll clear everything out and start fresh on the issue, keeping the changes small and focused to avoid any unintended effects. I’ll also make sure to update the tests so they run successfully. Thanks again for guiding me on this!

**Comment by AnyaVeselova** [comment]
Hi @nichgalzin, I'm stuck on this. The errors come from the profilePage component, which I haven't touched. I checked the DOM and components and there's this div.card with all the data. I can try to solve the issue without breaking the form into separate components and see if this issue still persists.

**Comment by AnyaVeselova** [comment]
Hi @nichgalzin, I finally gave up on refactoring the entire app in one go… Now all tests are passing 🎉 I think when you were trying to delete the prop during the meeting you just didn't delete all the instances maybe.

**Review by nichgalzin — APPROVED** [review]
This is great Anya, focused and specific and addresses just the issue at hand. Thanks very much for this.
